packages/web-components/src/components/va-pagination/va-pagination.tsx

```diff
@@ -0,0 +1,229 @@
+import { Component, Host, h, Prop, Fragment } from '@stencil/core';
+import classnames from 'classnames';
+
+declare const window: Window &
+  typeof globalThis & {
+    dataLayer: any;
+  };
+
+@Component({
+  tag: 'va-pagination',
+  styleUrl: 'va-pagination.css',
+  shadow: true,
+})
+export class VaPagination {
+  /**
+   * Fires when a page is selected
+   */
+  @Prop() pageSelect: any;
+
+  /**
+   * Fires when a page is selected if enable-analytics is true
+   */
+  @Prop() trackEvent: any;
+
+  /**
+   * Aria-label suffix text for buttons
+   */
+  @Prop() ariaLabelSuffix: string = '';
+
+  /**
+   * Whether or not an analytics event will be fired
+   */
+  @Prop() enableAnalytics: boolean = true;
+
+  /**
+   * The maximum number of pages to show at once
+   */
+  @Prop() maxPageListLength: number = 10;
+
+  /**
+   * The current page number
+   */
+  @Prop() page: number;
+
+  /**
+   * The total number of pages
+   */
+  @Prop() pages: number;
+
+  /**
+   * Display last page number when the page count exceeds
+   * `maxPageListLength`
+   */
+  @Prop() showLastPage: boolean;
+
+  private handleTrackEvent = (...args) => {
+    if (!this.trackEvent) {
+      if (!window.dataLayer) return;
+      window.dataLayer.push(...args);
+    } else {
+      this.trackEvent(...args);
+    }
+  };
+
+  private handlePageSelect = (page, eventID) => {
+    this.pageSelect(page);
+    if (!this.enableAnalytics) return;
+
+    this.handleTrackEvent({
+      'event': eventID,
+      'paginate-page-number': page,
+    });
```

Comment:
Here's an example of firing the analytics event:


https://github.com/department-of-veterans-affairs/component-library/blob/69655465824e556cc6ea5291be94557e61448986/packages/web-components/src/components/va-progress-bar/va-progress-bar.tsx#L36-L45